### PR TITLE
Remove torch specific Dataloader, remove `num_workers` from torch models.

### PR DESCRIPTION
**Pull request by jaheba**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

**Comment by lostella** [comment]
When setting `num_batches_per_epoch`, and iterating multiple times, this goes over the same data again and again. I think `iter` is missing somewhere. Example:

```python
from gluonts.dataset.loader import data_loader
import numpy as np

dataset = [{"a": np.array([k])} for k in range(100)]

batches = data_loader(dataset, batch_size=5, num_batches_per_epoch=3, output_type=np.array)
```

Iterate twice with

```python
for batch in batches:
    print(batch)
```

Outputs both times

```python
{'a': array([[0],
       [1],
       [2],
       [3],
       [4]])}
{'a': array([[5],
       [6],
       [7],
       [8],
       [9]])}
{'a': array([[10],
       [11],
       [12],
       [13],
       [14]])}
```

File: src/gluonts/torch/model/simple_feedforward/estimator.py [hunk 1]
```diff
@@ -207,27 +203,17 @@ def create_training_data_loader(
...
-            ),
-            self.num_batches_per_epoch,
+        return data_loader(
+            instances,
+            cycle=True,
+            batch_size=self.batch_size,
+            shuffle_buffer_length=shuffle_buffer_length,
+            field_names=TRAINING_INPUT_NAMES,
+            output_type=torch.tensor,
+            num_batches_per_epoch=self.num_batches_per_epoch,
```
Comment (lostella) [inline src/gluonts/torch/model/simple_feedforward/estimator.py]: Here, `Cyclic` is applied after instance splitting, while prior to the change it was applied before

**Comment by jaheba** [comment]
I think you need to use cycle in this case

**Comment by lostella** [comment]
> I think you need to use cycle in this case

That doesn't do it, `cycle` only makes sure the data is wrapped over when finished, but not that it takes off from where it left

File: src/gluonts/torch/model/simple_feedforward/estimator.py (hunk 1 above)
Comment (jaheba) [reply src/gluonts/torch/model/simple_feedforward/estimator.py]: I think it shouldn't matter, since we are not doing any caching.

Comment (lostella) [reply src/gluonts/torch/model/simple_feedforward/estimator.py]: There might be weird edge cases though: if the dataset has just one series, and the splitter happens to sample zero instances, then `Cyclic` will exhaust, see https://github.com/awslabs/gluonts/blob/c6be56264fefe5d879a1fe47ba57798097327533/src/gluonts/itertools.py#L77-L81

**Comment by jaheba** [comment]
> > I think you need to use cycle in this case
> 
> That doesn't do it, `cycle` only makes sure the data is wrapped over when finished, but not that it takes off from where it left

I see, you are right. The arg to `IterableSlice` needs to be wrapped in an `iter`.

File: src/gluonts/torch/model/simple_feedforward/estimator.py (hunk 1 above)
Comment (jaheba) [reply src/gluonts/torch/model/simple_feedforward/estimator.py]: Should we remove `cycle` from this then and ensure that the passed dataset is already cycled?

**Comment by lostella** [comment]
So maybe we need some good test here 😇

**Review by lostella — DISMISSED** [review]

**Review by lostella — APPROVED** [review]
